src/game/lara/lara_control.c

```diff
@@ -356,6 +359,8 @@ void Lara_HandleUnderwater(ITEM_INFO *item, COLL_INFO *coll)
 
     if (g_Lara.current_active && g_Lara.water_status != LWS_CHEAT) {
         Lara_WaterCurrent(coll);
+    } else {
+        LOT_ClearLOT(&g_Lara.LOT);
```

Comment:
Clearing it on every frame sounds like too much TBH. Can we be smarter about it?

Reply:
The weird thing is `g_Lara.current_active = 0;` happens every frame too in `Lara_WaterCurrent` which is why I did it this way. I'll try to think of something.

Reply:
walkawayy has asked about the fix over on discord, and for what its worth we've determined that this is what the later games that do not use LOT for currents do too, clear out all related data every frame

Reply:
I was thinking of clearing the LOT like this
```
if (g_Lara.current_active) {
    LOT_ClearLOT(&g_Lara.LOT);
    g_Lara.current_active = 0;
}
```
but IDK, if Troye says this fix is compatible with Tomb 2+ then I'm OK with the current variant.

Reply:
Where would that snippet be called? @Trxyebeep said he thinks the fix I implemented is fine.

Reply:
I meant to call it this way where currently the only assignment `g_Lara.current_active = 0;` is living, but I've no idea if it's even gonna work.

Reply:
This doesn't seem to work. Lara is slightly affected by current but it's jumpy and she can beat it easily by swimming.

https://streamable.com/8s2qpi


If you aren't comfortable with the `LOT_ClearLOT` I added, I can take them out. Those changes only fix a bug not present in the original games where Lara enters current 1 and travels to sink 2. Then, jumping into current 2 will never work. The main bug was an issue with wrongly restoring the LOT on load where it now clears.

Reply:
that looks as if its getting current for a frame and then not the next. i.e Clear, set , clear , set ...

Reply:
Was there a consensus on this? Is the current solution acceptable or do we want to keep exploring alternatives?

Reply:
Since my approach doesn't work, and besides my early optimization concerns, there are no problem with your solution, let's get this merged too.